ckan/templates/package/activity.html

```diff
@@ -8,5 +8,51 @@ <h1 class="hide-heading">
       {{ _('Activity Stream') }}
     {% endblock %}
   </h1>
-  {% snippet 'snippets/activity_stream.html', activity_stream=activity_stream, id=id, object_type='package' %}
+
+  {# allow extensions to continue using this template without new vars #}
+  {% if activity_types is defined %}
+  <div 
+    id="activity_types_filter"
+    style="margin-bottom: 15px;"
+    data-module="activity-stream"
+    >
+    <label for="activity_types_filter_select">Activity type</label>
+    <select id="activity_types_filter_select">
+      <option
+        {% if not activity_type %}selected{% endif %}
+        data-url="{{ h.url_for('dataset.activity', id=id) }}"
+      >{{ _('All activity types') }}</option>
+      {% for type_ in activity_types %}
+      <option 
+        {% if activity_type == type_ %}selected{% endif %}
+        data-url="{{ h.url_for('dataset.activity', id=id, activity_type=type_) }}"
+      >{{ type_.title() }}</option>
+      {% endfor %}
+    </select>
+
+  </div>
+  {% endif %}
+
+  {% if activity_stream|length > 0 %}
+    {% snippet 'snippets/activity_stream.html', activity_stream=activity_stream, id=id, object_type='package' %}
+  {% else %}
+    <p>
+      {{ _('No activity found') }}
+      {% if activity_type %}
+        {{ _('for this type') }}.
+      {% endif %}
+    </p>
+  {% endif %}
+  
+  <div id="activity_page_buttons" style="margin-top: 25px;">
+
+  {% if prev_page %}
+    <a href="{{ prev_page }}" class="btn btn-default btn-rounded" >{{ _('Previous page') }}</a>
+  {% endif %}
+
+  {% if next_page %}
+      <a href="{{ next_page }}" class="btn btn-default btn-rounded">{{ _('Next page') }}</a>
+  {% endif %}
```

Comment:
I found the "Previous page" and "Next page" buttons confusing, as when you click on previous you are actually requesting Newer (more recent) activities. Maybe is a bit more intuitive to use "Newer" and "Older" explicitly (and let's not use `btn-rounded` as it is not used elsewhere):

```suggestion
  {% if prev_page %}
    <a href="{{ prev_page }}" class="btn btn-default" >{{ _('Newer activities') }}</a>
  {% endif %}

  {% if next_page %}
      <a href="{{ next_page }}" class="btn btn-default">{{ _('Older activities') }}</a>
  {% endif %}
```